docs/watchlists-feeds-reports.rst

```diff
@@ -0,0 +1,244 @@
+
+Both IOCs and reports may include a ``link`` property, which is used by the Carbon Black Cloud console UI as a hint
+to indicate that this IOC or report is being managed outside of the console.  If this property is not ``None``,
+the console UI will disable the ability to edit the IOC or report, but they can still be edited via the API.
+
+Tips for Using IOCs
++++++++++++++++++++
+
+* You can safely ignore certain fields in an IOC.  For example, fields like ``alert_id`` and ``process_guid`` will
+  always uniquely identify just a single record in your organization's data, whereas a field like ``org_id`` will be
+  a constant across *all* your organization's data.
+* Timestamp fields such as ``backend_timestamp`` are useful in ad-hoc queries, to look for data occurring before or
+  after a certain date, but are of limited usefulness over the span of time a watchlist may be running.
+* A list of hashes (such as with ``process_sha256``) can be of limited value.  They are inconvenient to keep current,
+  especially as software (whether legitimate or malicious) gets updated over time, but are definitely easier to manage
+  with ``equality`` IOCs.
+* Counter fields (such as ``netconn_count``) can be useful with range queries to locate processes that are using a
+  large number of resources.  For example, the query ``netconn_count:[500 TO *]`` will match only processes that make
+  a large number of network connections.
+* When using ingress IOCs, be careful of errant characters in the ``values`` list, such as leading or trailing
+  whitespace or embedded newline characters.  These errant characters may cause the IOCs to fail, leading to false
+  negative results.
+* ``equality`` IOCs for IPv4 fields (e.g. ``netconn_remote_ipv4``) cannot support CIDR notation; full IP addresses
+  must be used.
+* ``equality`` IOCs for IPv6 fields (e.g. ``netconn_remote_ipv6``) do not support standard or CIDR notation at this
+  time. All IPv6 addresses must omit colon characters, spell out all zeroes in the address, and represent all
+  alphabetic characters in uppercase. For example, "ff02::fb" becomes "FF0200000000000000000000000000FB".
+* Query IOCs must always use field-prefixed queries (key-value pairs); they do not support just searching for a value
+  without a field specified.  Values in query clauses that do not specify fields will be ignored.
+
+:Wrong: ``process_name:chrome.exe AND 192.168.1.1``
+:Right: ``process_name:chrome.exe AND netconn_ipv4:192.168.1.1``
+
+* Query IOCs may search on CIDR address ranges, e.g.: ``netconn_ipv4:192.168.0.0/16``.
+
+Feeds
+-----
+Another way of managing reports is to attach them to a *feed.* Feeds can contain multiple reports, and a feed can be
+attached to a watchlist, effectively making the contents of the watchlist equivalent to the contents of the feed.
+
+A new feed may be created as follows:
+
+::
+
+    >>> from cbc_sdk.enterprise_edr import Feed, Report
```

Comment:
It's not my fault you have to use JSON instead of just creating objects in a reasonable fashion. That's part of what I want to address eventually.

That said, have you any suggestions here?

Reply:
I didn't mean it as in "don't blame me," I meant it as in "it is what it is." (At least until I get the chance to tear into it.)  Maybe just eliminating the report creation code (as it's covered adequately earlier) will make these easier to look at.

Reply:
I've slimmed down the examples by assuming the presence of a report, as noted in the text introducing each example.